### PR TITLE
fix: load/interface with GTM directly

### LINKED ISSUE #502: Load and interface with GTM directly
## What is the feature about
Removing reliance on a third-party to load GTM.

We are currently loading/interfacing with GTM via `react-gtm-module` but [due to security concerns](https://blog.kyber.network/notice-of-exploit-of-kyberswap-frontend-963aa8febd6a), we should load `gtm.js` and interface with it directly instead.

## The list of requirements
- Remove `react-gtm-module`
- Load `gtm.js` and intialise GTM environment
- Integrate direct interfacing in `trackEvent`

## Links
- [Slack thread](https://5afe.slack.com/archives/C03BHKS5K6D/p1661959521783399)
- [KyberSwap exploit](https://blog.kyber.network/notice-of-exploit-of-kyberswap-frontend-963aa8febd6a)
- [`react-gtm-module`](https://github.com/alinemorelli/react-gtm)
- [GTM installation guide](https://developers.google.com/tag-platform/tag-manager/web)
- Example [implementation](https://morganfeeney.com/how-to/integrate-google-tag-manager-with-next-js)
- [Use GTM with CSPs](https://developers.google.com/tag-platform/tag-manager/web/csp)

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #502

## How this PR fixes it

We no longer load GTM via a package. The syntax remains the same for initialization, wheras `TagManager.dataLayer` no longer requires nesting the event inside a `dataLayer` property.

## How to test it

- Open the Safe with Analytics cookies enabled. Observe that GTM loads as normal and events are tracked correctly.

## Analytics changes

GTM is loaded directly.

### CODE REVIEW

**Comment by iamacook** [comment]
I've left this as a draft because I'd like to review it in person.

**Review by usame-algan — APPROVED** [review]
Works great! 💯

File: src/services/analytics/TagManager.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  const noscript = document.createElement('noscript')
+
+  const gtmIframe = `
+      <iframe src="https://www.googletagmanager.com/ns.html?id=${gtmId}${auth}${preview}&gtm_cookies_win=x"
+        height="0" width="0" style="display:none;visibility:hidden" id="tag-manager"></iframe>`
+
+  noscript.innerHTML = gtmIframe
+
+  return noscript
+}
```
Comment (katspaugh) [inline src/services/analytics/TagManager.ts:71]: @iamacook our app doesn't work w/o JS, no need to include this.

Comment (iamacook) [reply src/services/analytics/TagManager.ts:71]: Fixed [here](https://github.com/safe-global/web-core/pull/538).

**Pull request merged**